### PR TITLE
Enable auto-approval on both channels for extensions in Notable group

### LINKED ISSUE #20991: Enable auto-approval on both channels for extensions in Notable group
Since https://mozilla-hub.atlassian.net/browse/AOP-133 we no longer want to pre-review all versions of extensions in the Notable group introduced in https://github.com/mozilla/addons-server/issues/19976 (further refined in https://github.com/mozilla/addons-server/issues/20351). Instead the versions will be flagged for human review and put in the manual review queue for post auto-approval review.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDSRV-455)

### PR DESCRIPTION
**Pull request by diox**
Fixes #20991

### CODE REVIEW

File: src/olympia/constants/promoted.py [hunk 1]
```diff
@@ -135,8 +135,8 @@ def __bool__(self):
     name=_('Notable'),
     api_name='notable',
     warning=False,
-    listed_pre_review=True,
-    unlisted_pre_review=True,
+    listed_pre_review=False,
+    unlisted_pre_review=False,
```
Comment (eviljeff) [inline src/olympia/constants/promoted.py:139]: `False` is the default so the lines can just be removed.

**Comment by diox** [comment]
Labelled as do not merge as we wait for confirmation whether or not to go through with this change.

**Review by eviljeff — APPROVED** [review]

File: src/olympia/constants/promoted.py [hunk 2]
```diff
@@ -1,48 +1,27 @@
...
-        False,  # flag_for_human_review - will be add-on be flagged for another review
-    ),
-)
-
-
-class PromotedClass(_PromotedSuperClass):
-    __slots__ = ()
+@dataclass
+class PromotedClass:
+    id: int = None
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/constants/promoted.py:10]: Not by choice :smile:  `@dataclass` wants types... I could have used `Any` but I figured I might as well put in real ones...

**Comment by diox** [comment]
We also need a data migration to flag existing versions - see https://github.com/mozilla/addons-server/issues/20991#issuecomment-1642428010

**Comment by diox** [comment]
Currently blocked so closing for now.

**Pull request closed without merging**